server/middleware/permissions/rulesClass.ts

```diff
@@ -51,17 +52,12 @@ export class Permissions {
     )
   }
 
-  /** Can create new report (ignoring caseload!) */
-  readonly canCreateReport: boolean
-
   private couldCreateReportInLocationIfActiveInService(code: string): boolean {
     return (
-      // has minimal roles
-      this.canCreateReport &&
       // if PECS region, can user edit PECS reports (has approver & PECS roles)?
-      ((isPecsRegionCode(code) && this.roles.has(roleApproveReject) && this.roles.has(rolePecs)) ||
-        // otherwise does user have caseload but isn't a data warden?
-        (this.caseloadIds.has(code) && !this.roles.has(roleApproveReject)))
+      (isPecsRegionCode(code) && this.roles.has(roleApproveReject) && this.roles.has(rolePecs)) ||
+      // otherwise does user have caseload but isn't a data warden?
+      (this.caseloadIds.has(code) && this.roles.has(roleReadWrite) && !this.roles.has(roleApproveReject))
```

Comment:
We could give this (and probably other conditions) a name, e.g. something like `this.isReportingOfficerIn(code)`.

Similarly, we could have a `this.isDataWarden()` and also `this.canReportAndApprovePecs(code)` or something like that.

I'm just thinking aloud here, but I do wonder if potentially these could be building blocks with nice names as I've seen more or less the same conditions in a few places.

PS: Do we need to check the absence of the `roleApproveReject` role? 🤔 I know there shouldn't be overlapping _**in theory**_ but I do wonder what it adds.

Reply:
But why would we want to break the experience for a user just because it has both hats? 🤔 

Somehow in my mind I see these roles as additive, e.g. you have the `readWrite` role and the prisonId is your caseload...shouldn't you then be able to create a report (don't matter if you also happen to have the `approveReject` role (even tho, granted a user should not have both in practice).

Reply:
As discussed offline, the main argument to have that additional check is a failsafe in case a user is accidentally (or else) be given both the readWrite role (RO) and the approveReject role (DW).

This kinda goes against the requirement of preventing someone from being both a Reporting Officer and a Data Warden as it would mean someone could both create a report and also approve it themselves.